packages/react/src/components/fileInput/FileInput.tsx

```diff
@@ -0,0 +1,554 @@
+import path from 'path';
+
+import React, { ChangeEvent, useEffect, useRef, useState } from 'react';
+
+// import core base styles
+import 'hds-core';
+import composeAriaDescribedBy from '../../utils/composeAriaDescribedBy';
+import classNames from '../../utils/classNames';
+import { Button } from '../button';
+import { IconPlus, IconPhoto, IconCross, IconDocument, IconUpload } from '../../icons';
+import { InputWrapper } from '../../internal/input-wrapper/InputWrapper';
+import styles from './FileInput.module.scss';
+
+type FileProperty = keyof File;
+
+const isEqualFileBy = (givenProperties: FileProperty[], a: File, b: File): boolean => {
+  const sameProps: FileProperty[] = givenProperties.filter((property: FileProperty) => a[property] === b[property]);
+  return sameProps.length === givenProperties.length;
+};
+
+const findDuplicateByNameAndType = (files: File[], fileToCompare: File): File | undefined =>
+  files.find((file: File) => isEqualFileBy(['name', 'type'], file, fileToCompare));
+
+type DragAndDropProps = {
+  label: string;
+  helperText: string;
+};
+
+type Language = 'en' | 'fi' | 'sv';
+
+type FileInputProps = {
+  /**
+   * The id of the input element
+   */
+  id: string;
+  /**
+   * The label for the input
+   */
+  label: string;
+  /**
+   * The label for the file button. The button is not visible for assistive technology
+   */
+  buttonLabel: string;
```

Comment:
In some components we have these props listed in alphabetical order, while in others they're not ordered like that. If I would have to guess based on few samples that I checked, I would say it's more common to have them in alphabetical order. So maybe we should order them here as well?

Reply:
- [ ] order props